docs/user/exercises/programming-exercise-AI-proof-guidelines.inc

```diff
@@ -0,0 +1,491 @@
+                  while (shoppingList[nextInShoppingList] == null) {
+                      nextInShoppingList++;
+                  }
+                  listWithoutNull[i] = shoppingList[nextInShoppingList++];
+              }
+              ...
+      }
+
+   .. note::
+      The java standard methods which can be used to solve specific
+      tasks need to be excluded in the test cases and feedback has to be provided to the student.
+      e.g. The method Arrays.sort is not allowed to be used.
+
+
+-  **Continuous feedback:** Exercise instructions should not be there to just give a
+   thorough step by step guide of the task. Have students deduce, using specific test cases,
+   what might be missing in the code.
+   This promotes critical thinking and resembles a more real world scenario:
+
+   -  null-checks
+
+   -  check existence of list
+
+   Example shows that AI disregards to check if shopping list exists:
+
+   Generated solution:
+
+   .. code:: java
+
+      public int search(String itemName) {
+         for (int i = 0; i < shoppingList
+                     .length; i++) {
+            if (shoppingList[i].getName()
+                     .equals(itemName)) {
+            return i;
+            }
+         }
+         return -1;
+      }
+
+   Example solution:
+
+   .. code:: java
+
+      public int search(String itemName) {
+      //check whether the shoppingList exists
+         if (!checkShoppingListExists()) {
+            return -1;
+         }
+
+         //the rest is the same as
+         //the AI solution
+
+      }
+
+-  **Split Up Instructions for Improved Engagement:** To prevent
+   students from blindly copying and pasting, it's essential to
+   distribute the instructions strategically. Ideally, around 80%
+   of the guidelines should be placed within the problem
+   statement, leaving a minimum of 20% to be incorporated
+   directly into the template code, marked clearly with "TODO"
+   annotations. This approach ensures students interact more
+   deeply with the code while also referencing the overarching
+   problem statement.
+
+   .. note::
+      You have to then write in the problem statement: "Also follow the TODOs
+      in the code."
+
+   The following table shows with a simple example, how that might look like in practice.
+
+.. list-table::
+   :widths: 50 50
+   :header-rows: 1
+
+   * - **Problem Statement**
+     - **Template Code**
+   * - Implement the `reverseString` method in the `StringUtil` class. Ensure the returned string is the reverse of the input string. Follow the TODOs in the code for more information.
+     -
+       .. code-block:: java
+
+          public class StringUtil {
+              public static String reverseString(String input) {
+                  // TODO: Return the reversed version of the input string.
+                  // Do not use built-in methods or libraries to reverse the string directly.
+                  return "";
+              }
+          }
```

Comment:
This table is pretty badly rendered on my device. Can you try to limit the width of the first column?
<img width="986" alt="image" src="https://github.com/ls1intum/Artemis/assets/18218285/7aa28b55-f1b2-454b-acc0-08c3d709b470">
<img width="989" alt="image" src="https://github.com/ls1intum/Artemis/assets/18218285/d216cd4e-5a0e-439b-9fc7-0fed041813b4">

Reply:
Got it. Thanks